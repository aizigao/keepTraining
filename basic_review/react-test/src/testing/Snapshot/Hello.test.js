import { unmountComponentAtNode, render } from "react-dom";
import { act } from "react-dom/test-utils";
import Hello from "./Hello";
import pretty from "pretty";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render targat
  container = document.createElement("div");
  document.body.append(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

// it("renders with or without a name", () => {
//   act(() => {
//     render(<Hello />, container);
//   });

//   expect(
//     pretty(container.innerHTML)
//   ).toMatchInlineSnapshot(); /* ... gets filled automatically by jest ... */

//   act(() => {
//     render(<Hello name="Jenny" />, container);
//   });

//   expect(
//     pretty(container.innerHTML)
//   ).toMatchInlineSnapshot(); /* ... gets filled automatically by jest ... */

//   act(() => {
//     render(<Hello name="Margaret" />, container);
//   });

//   expect(
//     pretty(container.innerHTML)
//   ).toMatchInlineSnapshot(); /* ... gets filled automatically by jest ... */
//   // expect(true).toBe(true);
// });

it("renders with or without a name", () => {
  act(() => {
    render(<Hello />, container);
  });

  expect(pretty(container.innerHTML)).toMatchInlineSnapshot(
    `"<span>Hey, stranger</span>"`
  ); /* ... gets filled automatically by jest ... */

  act(() => {
    render(<Hello name="Jenny" />, container);
  });

  expect(pretty(container.innerHTML)).toMatchInlineSnapshot(
    `"<h1>Hello, Jenny!</h1>"`
  ); /* ... gets filled automatically by jest ... */

  act(() => {
    render(<Hello name="Margaret" />, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot(); /* ... gets filled automatically by jest ... */
  // expect(true).toBe(true);
});
