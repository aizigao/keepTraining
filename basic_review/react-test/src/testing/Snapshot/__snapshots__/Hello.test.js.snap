// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with or without a name 3`] = `"<h1>Hello, Margaret!</h1>"`;

exports[`renders with or without a name: "<h1>Hello, Margaret!</h1>" 1`] = `"<h1>Hello, Margaret!</h1>"`;
